Casks/t/toneprint.rb

```diff
@@ -1,16 +1,27 @@
 cask "toneprint" do
-  version "4.5.11"
-  sha256 "f5f4a61d0ed5cce8db861b68638295d15a4600df312d05b9e0b4b595cadbf26c"
+  version "4.6.16,dBdeIScCJECmA6pTU4VfHA,bykSZtSgsUKVx3UABT5qSg,7+25e8c66a"
+  sha256 "d9bbfef769a14ccf2a024218efb7164420211c9d68f2630dd31ce4edbce2d106"
 
-  url "https://mediadl.musictribe.com/download/software/tcelectronic/TonePrintEditor/Toneprint_#{version}.dmg",
-      verified: "mediadl.musictribe.com/download/software/tcelectronic/TonePrintEditor/"
+  url "https://cdn.mediavalet.com/aunsw/musictribe/#{version.csv.second}/#{version.csv.third}/Original/Toneprint_#{version.csv.first}-public.#{version.csv.fourth}.dmg",
+      verified: "cdn.mediavalet.com/aunsw/musictribe/"
   name "TonePrint"
   desc "Alter the character of your TonePrint pedal"
   homepage "https://www.tcelectronic.com/tonePrints.html"
 
   livecheck do
     url "https://www.tcelectronic.com/.rest/musictribe/v1/downloadcenter/solr-dldatatable?brandName=tcelectronic&modelCode=P0CLC&iDisplayStart=0&iDisplayLength=15"
-    regex(/TonePrint[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
+    regex(%r{/musictribe/(\w+)/(\w+)/Original/TonePrint[._-]v?(\d+(?:\.\d+)+)[._-]public[._-](\d+\+\w{8})\.dmg}i)
```

Comment:
```suggestion
    regex(%r{/musictribe/(\w+)/(\w+)/Original/TonePrint[._-]v?(\d+(?:\.\d+)+)[._-]public[._-](\d+\+\h+)\.dmg}i)
```

This works for now. One suggested change is that the text after the `+` in the suffix is a hash (e.g., `7+25e8c66a`), so we should use `\h` to make the intention clear. With that change, I don't think there's any particular benefit to explicitly specifying the length and it would increase the chance of this regex breaking in the future, so it's fine to use `\h+` instead.